### DISCUSSION TITLE
Split Read/Write Traffic

### DISCUSSION THREAD

**Discussion by bestickley** [porsager/postgres#440] [Q&A] [answered]
Hello,

How can I split read/write traffic? Ideally, I'd like to be able to pass into the `postgres` function connection details for one writer instance and one or more reader instances and postgres (this library) could automatically chose between the connection details based upon the operations being performed (SELECT, INSERT, UPDATE).

Is this a feature request or does this already exist? I saw the [Multi Host Connection Details](https://github.com/porsager/postgres#multi-host-connections---high-availability-ha) portion of the docs but it doesn't really answer my question (I think).

For context, I'm using Amazon RDS Aurora Cluster with a reader instance and a writer instance.

**Comment by porsager** [comment, marked as answer]
I think you should create separate instances for that then. You can't decide from the operation alone if you're reading or writing, as you could easily have a mutating function like `select some_function_that_writes()` etc. 

So basically:
```js
const sql = postgres({ ...writableConnectionDetails })

const readOnly = postgres({ ...readOnlyConnectionDetails })
```
And then use the appropriate one where necessary.

**Reply by bestickley** [reply]
Make sense. Thanks @porsager!
